### PR TITLE
Setting up a development environment in a Dev Container

### PR DESCRIPTION
**Pull request by BrooklynDewolf**
## Changes introduced with this PR

* Included all the essential files to setup a Dev Container with Docker
* Included a guide `DEVELOPMENT.adoc` to setup a development environment in a Dev Container easily

Co-authored-by: Jean-Louis Dupond <jean-louis@dupond.be>

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "name": "oVirt",
+  "build": {
+    "dockerfile": "../Dockerfile"
+  },
+  "runArgs": ["--network=ovirt", "--ulimit=nofile=2048:2048"],
```
Comment (dupondje) [inline .devcontainer/devcontainer.json:25]: Document why the ulimit is needed

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+  "mounts": ["source=ovirt-pki,target=/home/build/ovirt/etc/pki/ovirt-engine,type=volume"],
+  "appPort": [ "8080:8080", "8443:8443", "8787:8787", "54323:54323", "9696:9696", "6642:6642", "35357:35357", "2222:2222", "6100:6100", "7410:7410"]
```
Comment (dupondje) [inline .devcontainer/devcontainer.json:28]: Might be nice to put in some comment for which service each port is needed?

File: DEVELOPMENT.adoc [hunk 3]
```diff
@@ -0,0 +1,75 @@
+# oVirt Engine - Dev Container setup & building
+
+Before beginning, ensure that [Docker](https://www.docker.com/) is installed on the machine that the Dev Container will be running on. This could be your local device or on a [remote machine](https://code.visualstudio.com/docs/devcontainers/containers#_open-a-folder-on-a-remote-ssh-host-in-a-container).
```
Comment (dupondje) [inline DEVELOPMENT.adoc]: s/Before beginning/Before starting/

File: DEVELOPMENT.adoc [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+### Setup the PostgresQL database
```
Comment (dupondje) [inline DEVELOPMENT.adoc]: PostgreSQL

File: DEVELOPMENT.adoc [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+### Building the application
+
+You can set up the development environment by executing the following command:
+
+```
+make install-dev PREFIX=/home/build/ovirt/
+```
+
+Running this command will initiate the application's build process. You can skip tests by appending the `SKIP_TESTS=1` flag to the previous command. Additionally, you can specify the installation directory using the `PREFIX` flag.
```
Comment (dupondje) [inline DEVELOPMENT.adoc]: env vars are always set at the beginning of the command normally :)
SO: `SKIP_TESTS=1 make install-dev PREFIX=/home/build/ovirt/`

File: DEVELOPMENT.adoc [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+### Uploading images via `ovirt-imageio`
+
+To enable image uploading and downloading through the administration portal, you must run the ovirt-imageio daemon. Execute the following command to do so:
+
+```
+ovirt-imageio --conf-dir /home/build/ovirt/etc/ovirt-imageio
+```
+
+> If encountering a certificate error during connection testing, you'll need to accept the certificate in your browser. You can either locate the URL by accessing your browser's network console, clicking the 'Test Connection' button, and visiting the URL showing the error, OR simply navigate to `https://HOSTNAME:54323/info/` and accept the certificate.
```
Comment (dupondje) [inline DEVELOPMENT.adoc]: Or you install the oVirt generated CA as a trusted CA in your browser

File: DEVELOPMENT.adoc [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+### Connecting via console to your VM instance
+
+To establish a connection to the VM console via noVCN, you'll need to initiate the websocket service:
```
Comment (dupondje) [inline DEVELOPMENT.adoc]: noVNC

File: DEVELOPMENT.adoc [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+```
+/home/build/ovirt/share/ovirt-engine/services/ovirt-websocket-proxy/ovirt-websocket-proxy.py start
+```
+
+Afterward, head to your newly created VM and select the 'v' arrow adjacent to the 'Console' button. Proceed to 'Console Options' and set 'Console Invocation' to 'noVCN'. Save these settings and click on the 'Console' button.
```
Comment (dupondje) [inline DEVELOPMENT.adoc]: Afterwards
noVNC

File: frontend/webadmin/modules/pom.xml [hunk 9]
```diff
@@ -43,7 +43,7 @@
     <gwt-plugin.extraJvmArgs>-Dgwt.jjs.permutationWorkerFactory=com.google.gwt.dev.ThreadedPermutationWorkerFactory -Dgwt.jjs.maxThreads=4 -Djava.io.tmpdir="${project.build.directory}/tmp" -Djava.util.prefs.systemRoot="${project.build.directory}/tmp" -Djava.util.prefs.userRoot="${project.build.directory}/tmp" -Djava.util.logging.config.class=org.ovirt.engine.ui.gwtaop.JavaLoggingConfig ${gwt.jvmArgs}</gwt-plugin.extraJvmArgs>
     <!-- Control target browsers for GWT compilation, maps to 'user.agent' deferred binding property in *.gwt.xml -->
     <!-- By default, compile for Firefox browser only, use 'all-user-agents' profile to compile for all browsers -->
-    <gwt.userAgent>gecko1_8</gwt.userAgent>
+    <gwt.userAgent>safari</gwt.userAgent>
```
Comment (dupondje) [inline frontend/webadmin/modules/pom.xml]: Unrelated, remove change

File: packaging/bin/pki-create-ca.sh
Comment (dupondje) [inline packaging/bin/pki-create-ca.sh:1]: Would put this change in another PR

**Review by dupondje — CHANGES_REQUESTED** [review]

**Comment by BrooklynDewolf** [comment]
All feedback should have been processed, looks good?
